controlpanel/frontend/views/dashboard.py

```diff
@@ -0,0 +1,303 @@
+
+    def post(self, request, *args, **kwargs):
+        self.perform_update(**kwargs)
+        return super().post(request, *args, **kwargs)
+
+
+@method_decorator(feature_flag_required("register_dashboard"), name="dispatch")
+class AddDashboardAdmin(UpdateDashboard):
+    model = Dashboard
+    permission_required = "api.add_dashboard_admin"
+
+    def perform_update(self, **kwargs):
+        dashboard = self.get_object()
+        user = get_object_or_404(User, pk=self.request.POST["user_id"])
+
+        dashboard.admins.add(user)
+        messages.success(self.request, f"Granted admin access to {user.name}")
+
+
+@method_decorator(feature_flag_required("register_dashboard"), name="dispatch")
+class RevokeDashboardAdmin(UpdateDashboard):
+    model = Dashboard
+    permission_required = "api.revoke_dashboard_admin"
+
+    def perform_update(self, **kwargs):
+        dashboard = self.get_object()
+        user = get_object_or_404(User, pk=kwargs["user_id"])
+
+        dashboard.admins.remove(user)
+        messages.success(self.request, f"Removed admin access from {user.name}")
+
+
+@method_decorator(feature_flag_required("register_dashboard"), name="dispatch")
+class DashboardCustomers(OIDCLoginRequiredMixin, PermissionRequiredMixin, DetailView):
+    context_object_name = "dashboard"
+    model = Dashboard
+    permission_required = "api.retrieve_dashboard"
+    template_name = "dashboard-customers-list.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        dashboard = self.get_object()
+
+        customers = dashboard.viewers.all()
+        paginator = Paginator(customers, 25)
+
+        context["errors"] = self.request.session.pop("customer_form_errors", None)
+        context["page_no"] = page_no = self.kwargs.get("page_no")
+        context["paginator"] = paginator
+        context["elided"] = paginator.get_elided_page_range(page_no)
+        context["customers"] = customers
+        context["remove_customer_form"] = RemoveCustomerByEmailForm()
+        return context
+
+
+@method_decorator(feature_flag_required("register_dashboard"), name="dispatch")
+class AddDashboardCustomers(OIDCLoginRequiredMixin, PermissionRequiredMixin, UpdateView):
+    model = Dashboard
+    form_class = AddCustomersForm
+    permission_required = "api.add_dashboard_customer"
+
+    def form_invalid(self, form):
+        self.request.session["customer_form_errors"] = form.errors
+        messages.error(self.request, "Could not add customer(s)")
+        return HttpResponseRedirect(self.get_success_url())
+
+    def form_valid(self, form):
+        self.get_object().add_customers(form.cleaned_data["customer_email"])
+        messages.success(self.request, "Successfully added customers")
+        return HttpResponseRedirect(self.get_success_url())
+
+    def get_form_kwargs(self):
+        kwargs = FormMixin.get_form_kwargs(self)
+        return kwargs
+
```

Comment:
```suggestion
```
As per comment above, if you inherit from the correct subclasses there is no need to override this method. 

The reason this was needed before (presume it was copied from the app views?) was that the `UpdateView` implementation of `get_form_kwargs` passes an `instance` kwarg to your form, but your form does not expect this since it is not a ModelForm, so would raise an error.